File: backend/src/engine/models/order.py
class Order:
    def __init__(self, user_id, symbol, order_type, quantity, price):
        self.user_id = user_id
        self.symbol = symbol
        self.type = order_type
        self.quantity = quantity
        self.price = price
        self.status = 'pending'

    def to_dict(self):
        return {
            'user_id': self.user_id,
            'symbol': self.symbol,
            'type': self.type,
            'quantity': self.quantity,
            'price': self.price,
            'status': self.status
        }

File: backend/src/models/Order.ts
import mongoose, { Schema, Document } from "mongoose";

export interface IOrder extends Document {
  userId: string;
  symbol: string;
  type: string;
  quantity: number;
  price: number;
  status: string;
  createdAt: Date;
}

const OrderSchema: Schema = new Schema({
  userId: { type: String, required: true },
  symbol: { type: String, required: true },
  type: { type: String, required: true },
  quantity: { type: Number, required: true },
  price: { type: Number, required: true },
  status: { type: String, default: "pending" },
  createdAt: { type: Date, default: Date.now },
});

export default mongoose.model<IOrder>("Order", OrderSchema);
